app-frontend/src/app/components/pages/project/navbar/index.js

```diff
@@ -0,0 +1,92 @@
+import _ from 'lodash';
+import tpl from './index.html';
+import {Set} from 'immutable';
+
+class ProjectLayersNavController {
+    constructor(
+        $rootScope, $state, $scope
+    ) {
+        'ngInject';
+        $rootScope.autoInject(this, arguments);
+    }
+
+    $onInit() {
+        this.navs = [];
+        this.$scope.$watch('$ctrl.$state.current', this.onStateCurrentChange.bind(this));
```

Comment:
I think it's more idiomatic to use the `$transition` service.
It's an api that was exposed in the update to ui-router 1.x, and doesn't require polling using $watch on every digest cycle. I'll probably be gradually moving other views over to it as well.
eg:
```javascript
$transition.onSuccess({}, (transition) => {
	let toState = transition.to();
    ...
}
```